docs/layers/tile-layer.md

```diff
@@ -171,35 +297,41 @@ Returns:
 An array of objects in the shape of `{x, y, z}`.
 
 
-### Render Options
+##### `getTileMetadata`
 
-##### `renderSubLayers` (Function, optional))
+`getTileMetadata({x, y, z})`
 
-Renders one or an array of Layer instances with all the `TileLayer` props and the following props:
+Generates additional metadata for a tile. The metadata is then stored as properties in a `tile` instance. The default implementation adds a `bbox`.
 
-* `id`: An unique id for this sublayer
-* `data`: Resolved from `getTileData`
-* `tile`: An object containing tile index `x`, `y`, `z`, and `bbox` of the tile.
+Receives arguments:
 
-- Default: `props => new GeoJsonLayer(props)`
+- `x` (Number)
+- `y` (Number)
+- `z` (Number)
 
+Returns:
 
-### Callbacks
+An object that contains the fields that should be added to the `tile` instance.
 
-##### `onViewportLoad` (Function, optional)
 
-`onViewportLoad` is a function that is called when all tiles in the current viewport are loaded. The loaded content (as returned by `getTileData`) for each visible tile is passed as an array to this callback function.
+##### `getParentIndex`
 
-- Default: `data => null`
+`getParentIndex({x, y, z})`
 
+Retrives the parent of a tile. The parent tile is usually a tile in the previous zoom level (`z - 1`) and contains the given tile. This method is called to populate the `parent` and `children` fields of a tile.
 
-##### `onTileError` (Function, optional)
+Returns:
 
-`onTileError` called when a tile failed to load.
+`{x, y, z}` of the parent tile.
 
-- Default: `console.error`
+
+##### `updateTileStates`
+
+`updateTileStates()`
```

Comment:
Nit: For the next release; I think this API (my understanding it requires calling getTileIndicesInViewport/updateTiles/selectedTiles) is a bit cumbersome. 

Also to support multi-viewport scenarios without thrashing I believe both this class and the Tileset3D should be able to accept a list of viewports when selecting tiles to be loaded and then return a list of tiles per viewport.